docs/USB Flashing.md

```diff
@@ -37,3 +34,173 @@ With the board connected and in bootloader mode (reset it by sending the charact
+udev rules can - among other things - change the ownership, the group ownership, and the permissions of the new USB device file. So, a udev rule can be used to grant full access to the new USB device to everyone in a Unix group.
+
+The rule needs to be smart - we do not want to have it applied to every USB device that is connected to the PC, like your USB stick or your mouse. It should only be applied to your FC. Luckily, this is possible, because USB devices tell the host computer a lot about themselves - they present things like their manufacturer and their product ID. So, the udev rule should apply only to Flight Controllers.
+
+ST Microelectronics, the manufacturer of the microcontroller on your FC, has the (internationally standardized) USB Vendor ID '0483'. 
+
+
+A udev rule which will put all STM microcontrollers into group "plugdev" can be established e.g. by 
+```
+(echo '# STM32 MCU)'
+ echo 'SUBSYSTEMS=="usb", ATTRS{idVendor}=="0483", MODE="0664", GROUP="plugdev"') | sudo tee /etc/udev/rules.d/91-vendorSTM-group-plugdev.rules > /dev/null
+```
+(if you have no luck with it, try the more generic `KERNEL=="ttyACM[0-9]", GROUP="plugdev"` which sets any new USB device that is a ttyACM to group plugdev without checking whether it is a STM microcontroller).
+Note you might need to reboot your system for the rule to take effect. Usually though, a reboot is not needed, stopping the udev system service and restarting it has the same effect:
+
+```
+$ sudo systemctl stop udev
+$ sudo systemctl start udev
+```
+Connect your FC to USB, and then check the permissions of the resulting device file (e.g. /dev/ttyACM0). If the udev rule is set up properly, this device file should be group "plugdev" with group permissions "rw". Of course, you must be member of group "plugdev" to take advantage of it.
+
+
+When looking into the output of 'dmesg' above, you will have noticed that when the Cleanflight Configurator program was flashing the firmware, the USB device 'STM32 virtual COM port' disappeared (device 18), and instead, a new device appeared - the device 19, "STM Bootloader". We can see that it has the idVendor 0483 and the idProduct df11. 
+Every STM32 microcontroller has a special bootloader mode (the famous 'unbrickable bootloader', called DFU mode for Device Firmware Update) that identifies itself with a product ID of 'df11'. 
+
+So, Cleanflight Configurator has made the FC to unplug itself (virtually) from your PC and to reconnect again as a different device!
+
+Since your FC can magically disguise as two different devices - a DFU bootloader device, which has commands to flash a firmware, and a serial interface, which can be used to speak the MultiWii protocol and talk to the Cleanflight Configurator (configuration section) - there might be two different Linux devices /dev/ttyXXX and /dev/ttyYYY corresponding to the two modes of your FC.
```

Comment:
That's got nothing to do with magic, so it should not be called 'magically disguise'.